src/models/Offer.js

```diff
@@ -1,21 +1,50 @@
-export default class Offer {
-  constructor ({id, email, city, formData, expirationDate, confirmed, createdDate, deleted, hashedToken}) {
-    this.id = id
-    this.email = email
-    this.city = city
-    this.formData = formData
-    this.expirationDate = expirationDate
-    this.confirmed = confirmed
-    this.createdDate = createdDate
-    this.deleted = deleted
-    this.hashedToken = hashedToken
-  }
+import mongoose from 'mongoose'
+import forms from './forms'
 
-  isExpired () {
-    return this.expirationDate <= Date.now()
+const offerSchema = mongoose.Schema({
+  email: {
+    type: String,
+    lowercase: true,
+    required: [true, 'Missing email']
+  },
+  city: {
+    type: String,
+    lowercase: true,
+    required: [true, 'Missing city']
+  },
+  expirationDate: {
+    type: Date,
+    required: [true, 'Missing expiration date']
+  },
+  confirmed: {
+    type: Boolean,
+    required: [true, 'Missing confirmed property'],
+    default: false
+  },
+  deleted: {
+    type: Boolean,
+    required: [true, 'Missing deleted property'],
+    default: false
+  },
+  createdDate: {
+    type: Date,
+    required: [true, 'Missing created date'],
+    default: Date.now
+  },
+  hashedToken: {
+    type: String,
+    required: [true, 'Missing hashed token'],
+    unique: true // Create a mongodb unique index
+  },
+  formData: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: Array.from(Object.keys(forms)), // Must be one of the forms defined in forms.js
```

Comment:
Can we link the formRef to the city?
If I understand it correctly, currently the city could be 'augsburg' and the formData-ref could be 'neuburg' right?

Reply:
This is just the schema definition; a list of all possible entries that could ever be in this field. I took care of the situation you described in createOffer(). Have a look at [this](https://github.com/Integreat/neuburg-backend/blob/a6b38f1a2e812c3e1933959bf1121e3a6ffe2df6/src/services/OfferService.js#L13 ) :)